packages/react-query/src/suspense.ts

```diff
@@ -41,7 +41,12 @@ export const shouldSuspend = (
     | undefined,
   result: QueryObserverResult<any, any>,
   isRestoring: boolean,
-) => defaultedOptions?.suspense && willFetch(result, isRestoring)
+) => {
+  if (defaultedOptions?.suspense && result.isPending && !isRestoring) {
+    return true
+  }
+  return defaultedOptions?.suspense && willFetch(result, isRestoring)
```

Comment:
I think we can just simplify this to:

```
return defaultedOptions?.suspense && result.isPending
```

because we really want to suspend every time when we have no data, which `isPending` guarantees.

the `!isRestoring` will have the same problem as the offline mode - while restoring from external storage, we would render the component with `undefined` ... that is desired for normal useQuery, but not for suspense

Reply:
OK on it.